src/software/ai/hl/stp/action/action_visitor.h

```diff
@@ -29,12 +29,12 @@ class ActionVisitor
      * @param action The action to visit
      */
 
-    virtual void visit(const ChipAction& action)          = 0;
-    virtual void visit(const DribbleAction& action)       = 0;
-    virtual void visit(const InterceptBallAction& action) = 0;
-    virtual void visit(const KickAction& action)          = 0;
-    virtual void visit(const MoveAction& action)          = 0;
-    virtual void visit(const MoveSpinAction& action)      = 0;
-    virtual void visit(const PivotAction& action)         = 0;
-    virtual void visit(const StopAction& action)          = 0;
+    virtual void visit(ChipAction& action)          = 0;
```

Comment:
just curious if there are any other visitors that currently implement `action_visitor` that we would still want to pass in a const reference to? Or rather we need a visitor that has permission to change state + another class that doesn't?

Reply:
resolved as part of another comment